src/sql/src/plan/transform_expr.rs

```diff
@@ -491,6 +577,35 @@ pub fn fuse_window_functions(
         }
     }
 
+    let is_value_or_agg_window_func_call = |scalar_expr: &HirScalarExpr| -> bool {
+        // Look for calls only at the root of scalar expressions. This is enough
+        // because they are always there, see 72e84bb78.
```

Comment:
Will this function fail-safe if this invariant is violated in the future? i.e does it simply end up not doing the optimization or will it lead to wrong results?

Reply:
Yes, if the invariant is violated, then `fuse_window_functions` would just not do the optimization.